api/service/project_service.go

```diff
@@ -58,7 +57,7 @@ func (ps projectsService) List(ctx context.Context, name string) (mlp.Projects,
 		}
 	}
 	// Refresh cache and try filtering
-	err := ps.refreshProjects(ctx)
+	err := ps.refreshProjects(ctx, name)
```

Comment:
When retrieving the projects list, use the name filter if supplied. Without the name filter, all the projects will be retrieved which may be wasteful if we are only interested in one.